imagetest/test_suites/packagevalidation/windows_version_test.go

```diff
@@ -235,23 +236,37 @@ func TestServicesState(t *testing.T) {
 		services = append(services, "google_osconfig_agent")
 
 	}
-
-	for _, service := range services {
-		out, err := utils.RunPowershellCmd(fmt.Sprintf("(Get-Service -Name %s -ErrorAction Stop).Status", service))
-		if err != nil {
-			t.Fatalf("Error getting service status: %v %v", err, out.Stderr)
-		}
-		status := strings.TrimSpace(out.Stdout)
-		out, err = utils.RunPowershellCmd(fmt.Sprintf("(Get-Service -Name %s -ErrorAction Stop).StartType", service))
-		if err != nil {
-			t.Fatalf("Error getting service starttype: %v %v", err, out.Stderr)
+	checkServices := func() []error {
+		var errs []error
+		for _, service := range services {
+			out, err := utils.RunPowershellCmd(fmt.Sprintf("(Get-Service -Name %s -ErrorAction Stop).Status", service))
+			if err != nil {
+				errs = append(errs, fmt.Errorf("Error getting service status: %v %v", err, out.Stderr))
+				return errs
+			}
+			status := strings.TrimSpace(out.Stdout)
+			out, err = utils.RunPowershellCmd(fmt.Sprintf("(Get-Service -Name %s -ErrorAction Stop).StartType", service))
+			if err != nil {
+				errs = append(errs, fmt.Errorf("Error getting service starttype: %v %v", err, out.Stderr))
+				return errs
+			}
+			startType := strings.TrimSpace(out.Stdout)
+			if status != "Running" {
+				errs = append(errs, fmt.Errorf("service %s has unexpected Status, got %s want Running", service, status))
+			}
+			if startType != "Automatic" {
+				errs = append(errs, fmt.Errorf("service %s has unexpected StartType, got %s want Automatic", service, startType))
+			}
 		}
-		startType := strings.TrimSpace(out.Stdout)
-		if status != "Running" {
-			t.Errorf("service %s has unexpected Status, got %s want Running", service, status)
+		return errs
+	}
+	for {
+		errs := checkServices()
+		if len(errs) == 0 {
+			break
 		}
-		if startType != "Automatic" {
-			t.Errorf("service %s has unexpected StartType, got %s want Automatic", service, startType)
+		if err := ctx.Err(); err != nil {
+			t.Fatalf("context expired before services reached expected state: %v", err)
```

Comment:
I wonder if it'll be better to format the `err` part of this message ourselves: something like including linebreaks between each error message in the array for better readability.

Reply:
Updated error message with last state mismatch details, one line each